### PR TITLE
Add Email Verification to Users API

### LINKED ISSUE #3685: Propose and implement methods to verify user email
~~- GET/PATCH/DELETE `/users/me`~~
- Endpoint to verify email (please don't follow the current implementation, first propose a standard implementation being used widely)

### PR DESCRIPTION
**Pull request by poush**
Fixes #3685
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Adds the support for user verification through email. The concept of generating token and it's verification is taken from previous implementations.
Also, adds the email helpers so that token can be sent.

@niranjan94 @SaptakS Please do check the token generation and verification process. I have used the same concept as it was there before decoupling.

### CODE REVIEW

File: app/api/users.py [hunk 1]
```diff
@@ -99,12 +101,18 @@ def before_create_object(self, data, view_kwargs):
             raise ConflictException({'pointer': '/data/attributes/email'}, "Email already exists")
 
     def after_create_object(self, user, data, view_kwargs):
+        s = get_serializer()
+        hash = s.dumps([user.email, str_generator()])
+        # link = make_fe_url(path='/email/verify?token={token}'.format(token=hash))
```
Comment (poush) [inline app/api/users.py]: This method is added into Orders API so if it gets merged I will update here accordingly and if it takes time to review then I will the method definition here in this PR

Comment (shubham-padia) [inline app/api/users.py]: Please add the function here also, rest looks good :)

File: app/api/helpers/files.py [hunk 2]
```diff
@@ -160,3 +161,13 @@ def create_save_image_sizes(image_file, image_sizes_type, unique_identifier=None
     }
 
     return new_images
+
+
+def make_fe_url(settings=None, path=None):
```
Comment (shubham-padia) [inline app/api/helpers/files.py]: can you make this func name more descriptive, also add some more info in the docstring

**Comment by niranjan94** [comment]
@shubham-padia the `frontend_url` was intentional. We would be having multiple clients yes. But there will be a primary frontend client attached to the server instance. So, that reference is stored here. We need that for a lot of reasons (mainly for sending emails). We need to send them an email with verification link, forgot password link, link to download tickets, link to view tickets. For all this, we can't send an API link. So, thats why we have the primary frontend stored.

**Comment by SaptakS** [comment]
@poush please resolve the merge conflicts.

**Pull request merged**